In [1]:
import pandas as pd 
from scipy.stats import pointbiserialr
from scipy.stats import pearsonr
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('general_data.csv')

In [3]:
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [4]:
df.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [5]:
df.Attrition.replace(to_replace = ('No','Yes'),value = (0,1),inplace = True)


In [6]:
stats, p = pearsonr(df.Attrition, df.Age)
print(stats, p)

-0.15920500686577976 1.996801615886744e-26


# Hypothesis 1

null hypothesis (H0)  = Age and attrition have no correlation .

alternate hypothesis (H1) =  Age and attrition have a correlation between them .

r = -0.15920500686577965

p = 1.996801615886744e-26

r is negative so there is a negative relation between Age and attrition 

and as p <0.05 so null hypothesis (H0) is rejected and we find that there is a negative correlation between attrition and age 

In [7]:
stats, p = pearsonr(df.Attrition, df.DistanceFromHome)
print(stats, p)

-0.009730141010179667 0.5182860428050771


# Hypothesis 2
null hypothesis (H0)  = Attrition  and distance from home  have no correlation .

alternate hypothesis (H1) =  Attrition  and distance from home have a correlation between them .

r = -0.00973014101017967

p = 0.5182860428050771

Here p is greater than 0.05 hence we accept the null hypothesis (H0) that there is no realtion between attrtion and distance from home

In [8]:
stats, p = pearsonr(df.Attrition, df.MonthlyIncome)
print(stats, p)

-0.031176281698115076 0.03842748490600132


# Hypothesis 3
null hypothesis (H0)  = Attrition  and monthly income   have no correlation .

alternate hypothesis (H1) =  Attrition  and monthly income have a correlation between them .

r = -0.03117628169811501

p = 0.03842748490600132

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between TRRITION AND MONTHLY INCOME.

r is -.0311 so we see there is negative relation between the two 


In [9]:
stats, p = pearsonr(df.Attrition, df.PercentSalaryHike)
print(stats, p)

0.032532594891053535 0.030743386433355353


# Hypothesis 4
null hypothesis (H0)  = Attrition and salary hike percentage have no correlation .

alternate hypothesis (H1) =  Attrition and salary hike percentage have a correlation between them .

r = 0.03253259489105349

p = 0.030743386433355353

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and salary hike percent .

r is 0.032 so we see there is positive  relation between the two

In [10]:
stats, p = pearsonr(df.Attrition, df.TrainingTimesLastYear)
print(stats, p)

-0.04943057624425515 0.00102470619153603


# Hypothesis 5
null hypothesis (H0)  = Attrition and TrainingTimesLastYear have no correlation .

alternate hypothesis (H1) =  Attrition and TrainingTimesLastYear have a correlation between them .

r = -0.049430576244255

p = 0.0010247061915365072

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and TrainingTimesLastYear .

r is -0.049 so we see there is negative relation between the two

In [11]:
stats, p = pearsonr(df.Attrition, df.YearsAtCompany)
print(stats, p)

-0.13439221398997758 3.1638831224870065e-19


# Hypothesis 6
null hypothesis (H0)  = Attrition and YearsAtCompany have no correlation .

alternate hypothesis (H1) =  Attrition and YearsAtCompany have a correlation between them .

r = -0.1343922139899772

p = 3.1638831224877484e-19

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsAtCompany .

r is -0.134 so we see there is negative relation between the two

In [12]:
stats, p = pearsonr(df.Attrition, df.YearsSinceLastPromotion)
print(stats, p)

-0.033018775142584444 0.028330336189402967


# Hypothesis 7
null hypothesis (H0)  = Attrition and YearsSinceLastPromotion have no correlation .

alternate hypothesis (H1) =  Attrition and YearsSinceLastPromotion have a correlation between them .

r = -0.03301877514258433

p = 0.028330336189396753

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsSinceLastPromotion .

r is -0.033 so we see there is negative relation between the two

In [13]:
stats, p = pearsonr(df.Attrition, df.YearsWithCurrManager)
print(stats, p)

-0.15619931590162808 1.7339322652900218e-25


# Hypothesis 8
null hypothesis (H0)  = Attrition and YearsWithCurrManager have no correlation .

alternate hypothesis (H1) =  Attrition and YearsWithCurrManager have a correlation between them .

-0.15619931590162847

1.7339322652896276e-25

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsWithCurrManager.

r is -0.156 so we see there is negative relation between the two